BlazarService/src/test/java/com/hubspot/blazar/util/RepositoryBuildLauncherTest.java

```diff
@@ -0,0 +1,129 @@
+import com.hubspot.blazar.base.RepositoryBuild;
+import com.hubspot.blazar.data.service.BranchService;
+import com.hubspot.blazar.data.service.DependenciesService;
+import com.hubspot.blazar.data.service.ModuleDiscoveryService;
+import com.hubspot.blazar.data.service.ModuleService;
+import com.hubspot.blazar.data.service.RepositoryBuildService;
+import com.hubspot.blazar.discovery.ModuleDiscovery;
+import com.hubspot.blazar.github.GitHubProtos.Commit;
+
+@RunWith(JukitoRunner.class)
+public class RepositoryBuildLauncherTest {
+
+  private final RepositoryBuildService repositoryBuildService = mock(RepositoryBuildService.class);
+  private final BranchService branchService = mock(BranchService.class);
+  private final ModuleService moduleService = mock(ModuleService.class);
+  private final DependenciesService dependenciesService = mock(DependenciesService.class);
+  private final ModuleDiscoveryService moduleDiscoveryService = mock(ModuleDiscoveryService.class);
+  private final ModuleDiscovery moduleDiscovery = mock(ModuleDiscovery.class);
+  private final GitHubHelper gitHubHelper = mock(GitHubHelper.class);
+
+  @Test
+  public void itUsesPreviousCommitWhenBuildIsPartOfInterProjectBuild() throws Exception {
+    GitInfo branch = new GitInfo(Optional.of(1), "git.example.com", "example", "repository", 1, "master", true, 0L, 1L);
+
+    String commitSha = "0000000000000000000000000000000000000000";
+    Commit commit = Commit.newBuilder().setId(commitSha).build();
+    CommitInfo commitInfo = new CommitInfo(commit, Optional.absent(), Collections.emptyList(), false);
+
+    RepositoryBuild previousBuild = RepositoryBuild.newBuilder(1, 1, RepositoryBuild.State.SUCCEEDED, BuildTrigger.forCommit(commitSha), BuildOptions.defaultOptions())
+        .setCommitInfo(Optional.of(commitInfo))
+        .build();
+    Optional<RepositoryBuild> previousBuildOptional = Optional.of(previousBuild);
+
+    BuildOptions ipbBuildOptions = new BuildOptions(ImmutableSet.of(1), BuildOptions.BuildDownstreams.NONE, false);
+    RepositoryBuild currentBuild = RepositoryBuild.queuedBuild(branch, BuildTrigger.forInterProjectBuild(1), 2, ipbBuildOptions);
+
+    RepositoryBuildLauncher launcher = new RepositoryBuildLauncher(repositoryBuildService, branchService, moduleService, dependenciesService, moduleDiscoveryService, moduleDiscovery, gitHubHelper);
+
+    doAnswer(invocation -> {
+      Commit currentCommit = (Commit) invocation.getArguments()[1];
+      Optional<Commit> previousCommit = (Optional<Commit>) invocation.getArguments()[2];
+      assertThat(currentCommit.getId()).isEqualTo(commitSha);
+      assertThat(previousCommit.isPresent()).isTrue();
+      assertThat(previousCommit.get().getId()).isEqualTo(commitSha);
+      return null;
+    }).when(gitHubHelper).commitInfoFor(any(), any(), any());
+
+    when(gitHubHelper.shaFor(any(), any())).thenThrow(new IllegalStateException("Previous build is present this should not be called"));
```

Comment:
This kind of indirect testing could be dangerous. If the code changes for any reason your can end up with a test that tests nothing. Prefer to use the assert to directly check the expected outcome of the method that you unit test like `CommitInfo commitInfo = launcher.calculateCommitInfoForBuild()` and then check the commitInfoDirectly. 
As a proof, I already proposed above to directly generate the CommitInfo for IPBs so the `gitHubHelper).commitInfoFor` will not run and the assertions will not happen.